File: Program diferents lenguages/Builder/HouseBuilderJava/HouseBuilder/src/HouseBuilder/HighEndHouseBuilder.java
/*
 * To change this license header, choose License Headers in Project Properties.
 * To change this template file, choose Tools | Templates
 * and open the template in the editor.
 */
package HouseBuilder;


public class HighEndHouseBuilder implements HouseBuilder{

	private House house;
	
	public HighEndHouseBuilder() {
		this.house = new House();
	}
	
	@Override
	public void buildBasement() {
		house.setBasement("Fully Furnished");
	}

	@Override
	public void buildRoof() {
		house.setRoof("Wooden Roof");
	}

	@Override
	public void buildStructure() {
		house.setStructure("Strongest");
	}

	@Override
	public void buildInterior() {
		house.setInterior("Good");
	}
	
	public House getHouse() {
		return this.house;
		
	}

}

File: Program diferents lenguages/Builder/HouseBuilderC#/HouseBuilder/HighEndHouseBuilder.cs
using System;
using System.Collections.Generic;
using System.Text;

namespace HouseBuilder
{
    class HighEndHouseBuilder : HouseBuilder
    {
        private House house;

        public HighEndHouseBuilder()
        {
            this.house = new House();  //composition
        }


        public void buildBasement()
        {
            house.setBasement("Indian Basement");
        }

        public void buildStructure()
        {
            house.setStructure("Super Single Deluxe");
        }

        public void buildRoof()
        {
            house.setRoof("Solar Tiles Roofing");
        }

        public void buildInterior()
        {
            house.setInterior("Contemporary Maharaja");
        }

        public House getHouse()
        {
            return this.house;
        }

    }
}
